crates/drainer/src/health_check.rs

```diff
@@ -0,0 +1,268 @@
+                    })?;
+
+                    logger::debug!("Database delete was successful");
+
+                    Ok::<_, HealthCheckDBError>(())
+                })
+            })
+            .await?;
+
+        Ok(())
+    }
+
+    async fn health_check_redis(&self, conf: &Settings) -> CustomResult<(), HealthCheckRedisError> {
+        let redis_conn = redis_connection(conf).await;
+
+        redis_conn
+            .serialize_and_set_key_with_expiry("test_key", "test_value", 30)
+            .await
+            .change_context(HealthCheckRedisError::SetFailed)?;
+
+        logger::debug!("Redis set_key was successful");
+
+        redis_conn
+            .get_key("test_key")
+            .await
+            .change_context(HealthCheckRedisError::GetFailed)?;
+
+        logger::debug!("Redis get_key was successful");
+
+        redis_conn
+            .delete_key("test_key")
+            .await
+            .change_context(HealthCheckRedisError::DeleteFailed)?;
+
+        logger::debug!("Redis delete_key was successful");
+
+        redis_conn
+            .stream_append_entry(
+                TEST_STREAM_NAME,
+                &redis_interface::RedisEntryId::AutoGeneratedID,
+                TEST_STREAM_DATA.to_vec(),
+            )
+            .await
+            .change_context(HealthCheckRedisError::StreamAppendFailed)?;
+
+        logger::debug!("Stream append succeeded");
+
+        let output = self
+            .redis_conn
+            .stream_read_entries(TEST_STREAM_NAME, "0-0", Some(10))
+            .await
+            .change_context(HealthCheckRedisError::StreamReadFailed)?;
+        logger::debug!("Stream read succeeded");
+
+        let (_, id_to_trim) = output
+            .get(TEST_STREAM_NAME)
+            .and_then(|entries| {
+                entries
+                    .last()
+                    .map(|last_entry| (entries, last_entry.0.clone()))
+            })
+            .ok_or(error_stack::report!(
+                HealthCheckRedisError::StreamReadFailed
+            ))?;
+        logger::debug!("Stream parse succeeded");
+
+        redis_conn
+            .stream_trim_entries(
+                TEST_STREAM_NAME,
+                (
+                    redis_interface::StreamCapKind::MinID,
+                    redis_interface::StreamCapTrim::Exact,
+                    id_to_trim,
+                ),
+            )
+            .await
+            .change_context(HealthCheckRedisError::StreamTrimFailed)?;
+        logger::debug!("Stream trim succeeded");
+
+        Ok(())
+    }
+}
+
+#[allow(clippy::enum_variant_names)]
+#[derive(Debug, thiserror::Error)]
+pub enum HealthCheckDBError {
+    #[error("Error while connecting to database")]
+    DBError,
+    #[error("Error while writing to database")]
+    DBWriteError,
+    #[error("Error while reading element in the database")]
+    DBReadError,
+    #[error("Error while deleting element in the database")]
+    DBDeleteError,
```

Comment:
Nit: Naming convention.

```suggestion
pub enum HealthCheckDbError {
    #[error("Error while connecting to database")]
    DbError,
    #[error("Error while writing to database")]
    DbWriteError,
    #[error("Error while reading element in the database")]
    DbReadError,
    #[error("Error while deleting element in the database")]
    DbDeleteError,
```